packages/components/src/FeatureSwitch/FeatureSwitch.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { ReactNode, useState } from "react";
+import { AnimatePresence, motion } from "framer-motion";
+import { Heading } from "../Heading";
+import { Text } from "../Text";
+import { Content } from "../Content";
+import { Switch } from "../Switch";
+import { Button } from "../Button";
+import { Emphasis } from "../Emphasis";
+import styles from "./FeatureSwitch.css";
+
+interface FeatureSwitchProps {
+  readonly children?: ReactNode | ReactNode[];
+
+  /**
+   * Feature description.
+   */
+  readonly description: string;
+
+  /**
+   * Defines if the feature should be ON or OFF by default.
+   */
+  readonly enabled?: boolean;
+
+  /**
+   * Feature title.
+   */
+  readonly title?: string;
+
+  /**
+   * Determines if a save indicator should show up when toggling the switch.
+   *
+   * @default false
+   */
+  readonly hasSaveIndicator?: boolean;
+
+  /**
+   * Callback when interacting with the switch.
+   *
+   * @param newValue
+   */
+  onSwitch?(newValue: boolean): void;
+
+  /**
+   * Callback when clicking with the edit button. This also determines if the
+   * edit button should show up in the UI.
+   */
+  onEdit?(): void;
+}
+
+export function FeatureSwitch({
+  children,
+  description,
+  enabled,
+  onEdit,
+  onSwitch,
+  hasSaveIndicator = false,
+  title,
+}: FeatureSwitchProps) {
+  const [savedIndicator, setSavedIndicator] = useState(false);
+  return (
+    <Content>
+      <div className={styles.container}>
+        <div className={styles.content}>
+          <Content>
+            {title && <Heading level={4}>{title}</Heading>}
+            <Text>{description}</Text>
+          </Content>
+        </div>
+        <div className={styles.action}>
+          <Switch
+            value={enabled}
+            onChange={handleSwitch}
+            ariaLabel={description}
+          />
+          <AnimatePresence>
+            {hasSaveIndicator && savedIndicator && (
+              <motion.div
+                className={styles.savedIndicator}
+                initial={{ y: -4, opacity: 0 }}
```

Comment:
Gonna extract this out on its own `Animate` component https://github.com/GetJobber/atlantis/pull/126/commits/b586aa7ac8d8d6460980b9ec5b9a01f912be0c1f